From 3af5a1e579e3324a13ba1f892c7befb3ab32d899 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 7 Mar 2019 21:56:57 -0800
Subject: [PATCH] pattern.c: Change string lenght parameter to be size_t in
 do_row()

This fixes below error on some architectures e.g. RISC-V

pattern.c:455:9: error: 'memset' specified size between 18446744071562067968 and 18446744073709551615 exceeds maximum object size 9223372036854775807 [-Werror=stringop-overflow=]                                                                     455 |         memset(deduced, DOT, (size_t)len);                                                                                   |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 pattern.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pattern.c b/pattern.c
index ffadd3f..4e5f187 100644
--- a/pattern.c
+++ b/pattern.c
@@ -428,7 +428,7 @@ static bool do_row(unsigned char *known, unsigned char *deduced,
                    unsigned char *row,
                    unsigned char *minpos_done, unsigned char *maxpos_done,
                    unsigned char *minpos_ok, unsigned char *maxpos_ok,
-                   unsigned char *start, int len, int step, int *data,
+                   unsigned char *start, size_t len, int step, int *data,
                    unsigned int *changed
 #ifdef STANDALONE_SOLVER
                    , const char *rowcol, int index, int cluewid
-- 
2.17.1

